Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp

```diff
@@ -211,6 +305,85 @@ void NetKDRequestDevice::LogError(ErrorType error_type, s32 error_code)
   m_scheduler_buffer[2] = Common::swap32(new_code);
 }
 
+NWC24::ErrorCode NetKDRequestDevice::KDCheckMail(u32* mail_flag, u32* interval)
+{
+  m_scheduler_buffer[4] = Common::swap32(2);
```

Comment:
Does this function need to hold the `m_scheduler_buffer_lock` for this and the write at the very bottom?

Reply:
It should. In the future, many other functions will write to that index, such as `KDDownload`, `KDSaveMail`, etc...

Reply:
Are you sure you want to hold the lock for the entire function? You're sending off a synchronous network request in here so this could potentially take a while to finish.

Also, do you know what the meaning of the value in `m_scheduler_buffer[4]` is? It strikes me a bit odd that you're setting this to 2 up here and then to 0 at the end of the function, does anyone else even have the chance to inspect this in the meantime if you hold the lock for the entire function?

Reply:
`m_scheduler_buffer[4]` is the current KD task that is running. If you look at [this forum page](https://forum.wii-homebrew.com/index.php/Thread/51738-Collecting-Error-Codes/?__cf_chl_tk=h7vPIIbrcGkXWLVXpRjGG_GzYf.VAOR0AlAbIXx6AaY-1684685151-0-gaNycGzNClA#post590284) posted by Wiimm, you can find all the other possible values.

`(0 = system, 1 = account, 2 = check, 3 = receive, 5 = send, 6 = save, 7 = DL, 9 = NWC24)`

Thinking it over, the lock should not be held for the entire function as `GetSchedulerStat` could be called when this is running.

Regarding your last point, I looked it over and realized I handled it wrong, The value of `m_scheduler_buffer[4]` should be reset to you on every exit point of the function.

Reply:
Ah, okay that makes sense. Yes, in that case you want to lock, write the 2, release the lock, and then at the end of the function re-lock, write the 0 and maybe result, and leave.

Reply:
Maybe also leave a comment about what those values mean (or make an enum for it).